JavaFXUserInterface/src/main/java/SystemsController.java

```diff
@@ -134,15 +139,54 @@ public void handle(MouseEvent mouseEvent) {
             }
         });
 
+        //Adding items to the choiceBox (drop down list)
+        sortSystem.getItems().add("Default");
+        sortSystem.getItems().add("Ascending RUL");
+        sortSystem.getItems().add("Descending RUL");
+        //Default Value
+        sortSystem.setValue("Default");
+        //Listener on the sort ChoiceBox. Depending on the sort selected, all systems panes are cleared and generated again
+        //with the appropriate sort applied.
+        sortSystem.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
+            @Override
+            public void changed(ObservableValue<? extends String> observableValue, String oldValue, String newValue) {
+                switch (newValue) {
+                    case "Ascending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Ascending RUL");
+                        }
+                        break;
+                    case "Descending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Descending RUL");
+                        }
+                        break;
+                    default:
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Default");
+                        }
+                        break;
+                }
+            }
+        });
+
     }
 
     /**
      * Creates elements that are in the scene so the data can be displayed.
      *
      * @author Jeff
      */
-    public void generateThumbnails() {
-        for (Asset system: systems) {
+    public void generateThumbnails(String sortSelected) {
+        ObservableList<Pane> boxes = FXCollections.observableArrayList();
+        //Based on the sort selected by the user, the appropriate list of Asset in the appropriate order is returned.
+        ObservableList<Asset> sortedSystems = sortSystems(sortSelected);
```

Comment:
is it possible to move the sorting outside of the generate thumbnails, sort the global variable systems in the on change and then call generate thumbnails. why? because this code doesn't sort the list view (even if you can do it with the columns) and it will resort for no reason every time generate thumbnail is called even if the sorting selected by the user has not changed

Reply:
I move the sorting outside generateThumbnails